articles/modules/ROOT/pages/protecting-against-cypher-injection.adoc

```diff
@@ -1,167 +1,503 @@
+Cypher Injection is a way for maliciously formatted input to jump out of its context, and by altering the query itself, hijack the query and perform unexpected operations on the database.
 
-This is a cousin to SQL injection, but affecting our Cypher query language.
+This is a cousin to SQL injection, but affects our Cypher query language.
 
-One of the best illustrations of injection attacks is from an XKCD comic featuring Little Bobby Tables:
+A great illustration demonstrating SQL injection is from the XKCD comic featuring Little Bobby Tables:
 
 [link=https://xkcd.com/327/]
 image::https://imgs.xkcd.com/comics/exploits_of_a_mom.png[Little Bobby Tables]
 
-The boy's mother named him `Robert'; DROP TABLE STUDENTS;--`, ensuring that if his name was appended into a SQL statement that
-didn't clean their inputs or protect against an injection attack, the quote would end the context of a quoted name, the semicolon would end that statement,
-the STUDENTS table would be dropped, and the `--` would turn the remaining part of the query into a comment, so the remaining query could be ignored and avoid any syntax errors.
+The comic shows a mother who has named her child `Robert'; DROP TABLE STUDENTS;--`, ensuring that if his name was appended to an SQL statement,
+through string concatenation with no protections against injection attacks, that the additional quote would close the opened quote from the application,
+the semicolon would finish the statement, and the `--` would comment out the rest of the query, so that syntax errors could be avoided.
 
-This syntax is of course specific to SQL. Given that Bobby's mother would likely have continued her crusade, she might have had another child specifically for targeting Cypher, used by the leading graph database Neo4j.
-Lets call him Little Robby Labels, to differentiate him from his older brother:
+Given that Bobby’s mother would likely have continued her crusade, her next child might be named to target Cypher, used by the leading graph database Neo4j.
+Let’s call them Little Robby Labels.
 
-`"Robby' WITH DISTINCT true as haxxored MATCH (s:Student) DETACH DELETE s //"`
+`"Robby' WITH DISTINCT true as haxxored MATCH (s:Student) DETACH DELETE s; //"`
 
-This is an attempt at Cypher injection, the equivalent of the SQL injection attack, except the deletion of all :Student nodes happens within the same statement instead of splitting it into a separate statement.
-(The WITH clause in the middle is just to ensure we reduce down to a single row of input before matching to deleting all students, it's for efficiency)
+This is an attempt at Cypher Injection, the equivalent of the SQL Injection attack, but with the deletion of all :Student nodes instead.
+The `WITH` clause in the middle is to ensure the result set is reduced to a single row before deleting all students.
```

Comment:
it's not the result set, it's the intermediate cardinality that's reduced

Reply:
I would perhaps consider replacing the example with something simpler, and then just explain that the with statement is there to more easily append and insert our injection snippet. 
```
`"Robby' WITH true AS n MATCH (s:Student) DETACH DELETE s; //"`
```

Not sure how much value is adds to introduce the concept of reducing intermediate record sets to begin with 🤔

Reply:
Okay, I think I'll simplify it as you suggest Daniel :)